ci/test/pipeline.template.yml

```diff
@@ -134,6 +134,7 @@ steps:
 
   - id: cargo-test
     label: Cargo test
+    depends_on: build-x86_64
```

Comment:
@benesch needs to sign off on this, as he is the one who made cargo-test self-contained so that it can start immediately without waiting for the build to finish.

Reply:
I checked out the commit originally, and saw that the main motivation was not to upload/download huge test artifacts, which this isn't adding back.

Reply:
ok , fair enough, thanks!

Reply:
This was important! Unless I'm wildly off base, building `clusterd` takes hardly any time on top of building all the test binaries. You've got to build 100% of the crate graph in order to build the test binaries, so linking one additional binary doesn't really move the needle. It's also a lot less fiddly. 

I put up https://github.com/MaterializeInc/materialize/pull/19538 to revert this part of the PR, while _unreverting_ the part to use multiple test threads. I think that will get the 14m -> 9m speedup without the bugs! :D